probe/docker/registry.go

```diff
@@ -56,11 +56,13 @@ type Client interface {
 	ListImages(docker_client.ListImagesOptions) ([]docker_client.APIImages, error)
 	AddEventListener(chan<- *docker_client.APIEvents) error
 	RemoveEventListener(chan *docker_client.APIEvents) error
+
 	StopContainer(string, uint) error
 	StartContainer(string, *docker_client.HostConfig) error
 	RestartContainer(string, uint) error
 	PauseContainer(string) error
 	UnpauseContainer(string) error
+	AttachToContainer(docker_client.AttachToContainerOptions) error
```

Comment:
We could do that on top of everything we have, but we'll always need to deal with the case of the UI going away with an open pipe.  I think I'd prefer to leave it as the generic mechanism for now.